pkg/common-controller/snapshot_controller.go

```diff
@@ -79,6 +80,8 @@ const snapshotAPIGroup = crdv1.GroupName
 
 const controllerUpdateFailMsg = "snapshot controller failed to update"
 
+var ctx context.Context = context.Background()
```

Comment:
We should probably use the configured timeouts for all these

Reply:
sure. as discussed I will start with context.TODO(), then a follow up PR to configurable timeout .. thanks

Reply:
maybe make more sense to have it defined in snapshot_controller_base.go?

Reply:
for consistency and first pass, across side car repos we are introducing context.TODO() . the improvements can be done in a follow up PR accordingly.  Update the patchset which did migrate from context.Background() to .TODO() @yuxiangqian